lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -154,13 +154,9 @@ def add_customer_data(post, options, payment = nil)
         post['order']['companyInformation'] = {
           'name' => options[:company]
         }
-        post['order']['contactDetails'] = {
-          'emailAddress' => options[:email]
-        }
+        post['order']['contactDetails']['emailAddress'] = options[:email] if options[:email]
```

Comment:
Do we know that `post['order']['contactDetails']` will always be defined here? I'm wondering if it might be worth adding `post['order']['contactDetails'] ||= {}` before this line. That might be redundant though.

Reply:
Okay, [`nestable_hash`](https://github.com/activemerchant/active_merchant/pull/2915/files/d83f99dc537c52e6283d99194d4c514bf046ab71#diff-476876f224070b3b093c610adc75d562R321) makes this work, so ship it!